tests/providers/google/cloud/hooks/test_dataprep.py

```diff
@@ -0,0 +1,97 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from unittest import mock
+
+import pytest
+from mock import patch
+from requests import HTTPError
+from tenacity import RetryError
+
+from airflow.providers.google.cloud.hooks import dataprep
+
+JOB_ID = 1234567
+URL = "https://api.clouddataprep.com/v4/jobGroups"
+TOKEN = "1111"
+EXTRA = {"token": TOKEN}
+
+
+@pytest.fixture(scope="class")
+def mock_hook():
+    with mock.patch("airflow.hooks.base_hook.BaseHook.get_connection") as conn:
+        hook = dataprep.GoogleDataprepHook(dataprep_conn_id="dataprep_conn_id")
+        conn.return_value.extra_dejson = EXTRA
+        yield hook
+
+
+class TestGoogleDataprepHook:
+    def test_get_token(self, mock_hook):
+        assert mock_hook._token == TOKEN
+
+    @patch("airflow.providers.google.cloud.hooks.dataprep.requests.get")
+    def test_mock_should_be_called_once_with_params(self, mock_get_request, mock_hook):
+        mock_hook.get_jobs_for_job_group(job_id=JOB_ID)
+        mock_get_request.assert_called_once_with(
+            f"{URL}/{JOB_ID}/jobs",
+            headers={
+                "Content-Type": "application/json",
+                "Authorization": f"Bearer {TOKEN}",
+            },
+        )
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[HTTPError(), mock.MagicMock()],
+    )
+    def test_should_pass_after_retry(self, mock_get_request, mock_hook):
+        mock_hook.get_jobs_for_job_group(JOB_ID)
+        assert mock_get_request.call_count == 2
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[mock.MagicMock(), HTTPError()],
+    )
+    def test_should_not_retry_after_success(self, mock_get_request, mock_hook):
+        mock_hook.get_jobs_for_job_group.retry.sleep = mock.Mock()  # pylint: disable=no-member
+        mock_hook.get_jobs_for_job_group(JOB_ID)
+        assert mock_get_request.call_count == 1
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[
+            HTTPError(),
+            HTTPError(),
+            HTTPError(),
+            HTTPError(),
+            mock.MagicMock(),
+        ],
+    )
+    def test_should_retry_after_four_errors(self, mock_get_request, mock_hook):
+        mock_hook.get_jobs_for_job_group.retry.sleep = mock.Mock()  # pylint: disable=no-member
+        mock_hook.get_jobs_for_job_group(JOB_ID)
+        assert mock_get_request.call_count == 5
+
+    @patch(
+        "airflow.providers.google.cloud.hooks.dataprep.requests.get",
+        side_effect=[HTTPError(), HTTPError(), HTTPError(), HTTPError(), HTTPError()],
+    )
+    def test_raise_error_after_five_calls(self, mock_get_request, mock_hook):
```

Comment:
Should  we mention the number of retries and exponential back-off in the docs somewhere?  I am not sure about that one. I think we are not mentioning it elsewhere (assuming that the hard-coded values are reasonable). But maybe (possibly in a separate PR) we could make it more explicit on the typical retry scenarios we use for GCP classes? Or maybe we should make it somehow configurable for all GCP classes ? WDYT @turbaszek @mik-laj ?

Reply:
I did not mention anywhere else that tenacity retry has 5 calls. This case is only my interpretation. I was suggesting how it was done already done in airflow -- I mean that `num_retries`. I am open for suggestions.

Reply:
+1 for making the number of retries configurable